src/Sarif.Viewer.VisualStudio/TelemetryProvider.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.ApplicationInsights;
+using Microsoft.CodeAnalysis.Sarif.Converters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Sarif.Viewer
+{
+    /// <summary>
+    /// Wrapper class for Application Insights client.
+    /// </summary>
+    internal static class TelemetryProvider
+    {
+        private static TelemetryClient appInsightsClient;
+        private static readonly object lockObject = new object();
+        private static bool isInitialized = false;
+
+        /// <summary>
+        /// Initializes the static instance of the TelemetryProvider.
+        /// </summary>
+        /// <param name="telemetryKey">The instrumentation key of the target AI instance.</param>
+        public static void Initialize(string instrumentationKey)
+        {
+            lock (lockObject)
+            {
+                if (!isInitialized)
+                {
+                    appInsightsClient = new TelemetryClient()
+                    {
+                        InstrumentationKey = instrumentationKey
+                    };
+
+                    isInitialized = true;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sends event telemetry when the user chooses a menu item from the Tools > Open Static Analysis Log File menu.
+        /// </summary>
+        /// <param name="toolFormat">The tool format of the menu item.</param>
+        public static void WriteMenuCommandEvent(string toolFormat)
+        {
+            if (string.IsNullOrWhiteSpace(toolFormat))
+            {
+                throw new ArgumentNullException(nameof(toolFormat));
+            }
+
+            WriteEvent(TelemetryEvent.LogFileOpenedByMenuCommand, "Format".KeyWithValue(toolFormat == ToolFormat.None ? "SARIF" : toolFormat));
```

Comment:
>ToolFormat.None [](start = 102, length = 15)

Is ToolFormat.None really a valid value to apss here? Or are we masking some other issue if this code path is exercised?

Reply:
Yeah if you look at OpenLogFileCommands.cs line 168, you see that the tool format is left at None for sarif files.